sdk/search/search-documents/src/searchIndexingBufferedSender.ts

```diff
@@ -0,0 +1,193 @@
+import {SearchClient} from "./searchClient";
+import {IndexDocumentsBatch} from "./indexDocumentsBatch";
+import {IndexDocumentsAction, SearchIndexingBufferedSenderOptions, SearchIndexingBufferedSenderUploadDocumentsOptions, SearchIndexingBufferedSenderMergeDocumentsOptions, SearchIndexingBufferedSenderMergeOrUploadDocumentsOptions, SearchIndexingBufferedSenderDeleteDocumentsOptions, SearchIndexingBufferedSenderFlushDocumentsOptions} from "./indexModels";
+import {IndexDocumentsResult} from "./generated/data/models";
+import {RestError, OperationOptions} from "@azure/core-http";
+import EventEmitter from 'events';
+import {createSpan} from "./tracing";
+import {CanonicalCode} from "@opentelemetry/api";
+
+const DEFAULT_BATCH_SIZE:number = 1000;
+const DEFAULT_FLUSH_WINDOW:number = 60000;
+const RETRY_COUNT:number = 3;
+
+/**
+ * Class used to perform buffered operations against a search index,
+ * including adding, updating, and removing them.
+ */
+export class SearchIndexingBufferedSender<T> {
+  private client:SearchClient<T>;
+  private autoFlush:boolean;
+  private flushWindowInMs:number;
+  private batchSize:number;
+  private batchObject: IndexDocumentsBatch<T>;
+  private cleanupTimer?: () => void;
+  private readonly emitter = new EventEmitter();
+
+  constructor(client: SearchClient<T>, options: SearchIndexingBufferedSenderOptions = {}) {
```

Comment:
Is there a way to remove the access to the constructor to public, but still be accessible from SearchClient?

Reply:
I think you probably want to use the pattern I used in `core-https` for this, where you make the public type an interface, and have a private class that implements it. Here's an example:

https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/core/core-https/src/httpHeaders.ts

That way you're just returning them a `SearchIndexingBufferedSender<T>` but the concrete class is hidden and can't be constructed easily.